ai-quick-actions/ssh-notebook-session.md

```diff
@@ -0,0 +1,199 @@
+
+# SSH Data Science Notebook Session
+
+## Table of Contents
+  - [Introduction](#introduction)
+  - [Prerequisite](#prerequisite)
+  - [Step-by-Step Guide](#step-by-step-guide)
+    - [Reveal the Notebook Private IP](#reveal-the-notebook-private-ip)
+    - [Run the Script](#run-the-script)
+    - [Enable OpenSSH Server Script](#enable-openssh-server-script)
+    - [Run the `enablessh.sh` Script](#run-the-enablesshsh-script)
+    - [Store the Private Key](#store-the-private-key)
+    - [Create Bastion](#create-bastion)
+    - [Create Bastion Session](#create-bastion-session)
+    - [Connect to the Bastion](#connect-to-the-bastion)
+    - [Tunnel](#tunnel)
+    - [VSCode Configuration](#vscode-configuration)
+
+## Introduction
+Enabling the SSH tunnel to OCI Data Science notebooks provides the ability to do remote debugging and coding using tools like VSCode Remote-SSH or PyCharm's equivalent feature. Additionally, you can connect to the instance for manual debugging.
+
+## Prerequisite
+- Create a [Notebook Session](https://docs.oracle.com/en-us/iaas/data-science/using/manage-notebook-sessions.htm) in OCI Data Science Service using your own [VCN](https://docs.oracle.com/en-us/iaas/Content/Network/Tasks/VCNs.htm) and [Private Subnet](https://docs.oracle.com/en-us/iaas/Content/Network/Concepts/privateaccess.htm).
+
+## Step-by-Step Guide
+This guide outlines the steps required to enable SSH tunneling.
+
+### Reveal the Notebook Private IP
+Create a file named `nb_ip.py` and add the following Python code to reveal the private IP address of the notebook:
+
+```python
+import oci
+import os
+
+print(f"- get NB_SESSION_OCID")
```

Comment:
Lint'er flags this since f-string not needed without substitution variables - extremely minor though, but ruff would object and fix it